src/pages/dashboard/index.js

```diff
@@ -10,25 +10,34 @@ import "./style.css";
 import useAuth from "../../hooks/useAuth";
 
 const Dashboard = () => {
+  const [searchVal, setSearchVal] = useState("");
+  const [searchResults, setSearchResults] = useState([]); // State for search results
+  const [isLoading, setIsLoading] = useState(false); // Loading state
+  const [errorMessage, setErrorMessage] = useState(""); // Error message state
+
 	const [searchVal, setSearchVal] = useState('');
 	 const { user } = useAuth();
 
-	const onChange = (e) => {
-		setSearchVal(e.target.value);
-	};
 
-	// Use the useModal hook to get the openModal and setModal functions
-	const { openModal, setModal } = useModal();
+  const onChange = (e) => {
```

Comment:
onChange what? can we rename to `onChangeSearchTerm` or something like this for better clarity?